cpp/oneapi/dal/algo/pca/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -39,25 +39,65 @@ static result_t infer(const context_gpu& ctx, const descriptor_t& desc, const in
     const auto data = input.get_data();
     auto model = input.get_model();
     auto eigenvectors = model.get_eigenvectors();
+    auto means = model.get_means();
+    auto eigenvalues = model.get_eigenvalues();
+
     const std::int64_t row_count = data.get_row_count();
+    const std::int64_t col_count = data.get_column_count();
```

Comment:
I suggest to cover this code with additional checks e.g.
ONEDAL_ASSERT(data.get_column_count() > 0);